client/extensions/wp-super-cache/cache-stats.jsx

```diff
@@ -2,49 +2,97 @@
  * External dependencies
  */
 import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
+import { flowRight, get } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import Button from 'components/button';
 import FoldableCard from 'components/foldable-card';
+import { errorNotice, removeNotice } from 'state/notices/actions';
+import { deleteFile } from './state/stats/actions';
+import { hasFileDeleteError, isDeletingFile } from './state/stats/selectors';
 
-function getAge( { lower_age, upper_age } ) {
-	if ( lower_age && upper_age ) {
-		return `${ lower_age } - ${ upper_age }`;
-	} else if ( lower_age ) {
-		return lower_age;
-	} else if ( upper_age ) {
-		return upper_age;
+function getAge( lower, upper ) {
+	if ( lower && upper ) {
+		return `${ lower } - ${ upper }`;
+	} else if ( lower ) {
+		return lower;
+	} else if ( upper ) {
+		return upper;
 	}
 
 	return '';
 }
 
 class CacheStats extends Component {
 	static propTypes = {
+		deleteFile: PropTypes.func.isRequired,
+		errorNotice: PropTypes.func.isRequired,
 		files: PropTypes.array,
+		hasError: PropTypes.bool,
 		header: PropTypes.string,
+		isDeleting: PropTypes.bool,
+		removeNotice: PropTypes.func.isRequired,
+		siteId: PropTypes.number.isRequired,
 		translate: PropTypes.func.isRequired,
 	};
 
 	static defaultProps = {
 		files: [],
+		hasError: false,
+		header: '',
+		isDeleting: false,
 	};
 
+	state = {
+		url: '',
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.isDeleting || ! prevProps.isDeleting || ! this.props.hasError ) {
+			return;
+		}
+
+		this.props.errorNotice(
+			this.props.translate( 'There was a problem deleting the cached file. Please try again.' ),
+			{ id: 'wpsc-cached-file-delete' }
+		);
```

Comment:
Same as https://github.com/Automattic/wp-calypso/pull/13923#issuecomment-301489315 -- you might be able to ditch the `componentDidUpdate` method (and some action props) if you handle notices by dispatching their corresponding actions directly from the `deleteFile` action.